enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+- User-defined tags can be updated on the following AWS resources.
+  1. EC2 instances for master and worker nodes.
+  2. Image registry.
+  3. Ingress LB.
+  4. IAM credentials by CCO in mint mode of operation.
 
 ## Proposal
 
-New `experimentalPropagateUserTags` field added to `.platform.aws` of install config to indicate that the user tags should be applied to AWS
-resources created by in-cluster operators.
+- Existing `experimentalPropagateUserTags` will be renamed to `propagateUserTags` in `.platform.aws` of install config to indicate that the user tags should be applied to AWS
+resources created by in-cluster operators.\
+  `experimentalPropagateUserTags` field will be set for deprecation.
+
+  If `propagateUserTags` is set to true, install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io`.
+
+- Add a new field `resourceTags` to `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type. Tags included in the
+  `resourceTags` field will be applied to new resources created for the cluster. The `resourceTags` field will be populated by the installer only if the `propagateUserTags` field is true.
+
+  `.spec.platformSpec.aws` is a mutable field and `.status.platformStatus.aws` is immutable.
+  `.status.platformStatus.aws` will have older version tags defined and is required for upgrade case.
+
+- All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api) will apply these tags to all AWS resources they create.\
+  Operator must update the AWS resource within (5 minutes + cloud provide API response time).\
+  Operator must update AWS resource to match requested user-defined tags provided in `.spec.platformSpec.aws` (or `.status.platformStatus.aws`).\
+  Operator must consider `.status.platformStatus.aws` when AWS resource is created. `.status.platformStatus.aws` can be ignored for user-defined tag update/delete.
+
+### Create tags scenarios
+`resourceTags` that are specified in  `.spec.platformSpec.aws` of the Infrastructure resource will merge with user-defined tags in an individual component.
+
+In the case where a user-defined tag is specified in the Infrastructure resource and
+1) There is already user tag with the same key present for AWS resource, the value from the AWS resource will be replaced.\
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value_comp1`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`\
+   Action = Existing tag for AWS resource is updated to reflect new value.\
+   Final tag set to AWS resource = `key_infra1 = value_infra1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+2) There is no tag with same key present, new user-defined tag is created for the AWS resource. In case of limit reached, a validation error is generated.\
```

Comment:
```suggestion
2) There is no tag with same key present, new tag is created for the AWS resource. In case of limit reached, a validation error is generated.\
```